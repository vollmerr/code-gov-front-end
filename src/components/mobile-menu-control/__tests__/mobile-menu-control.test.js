import React from 'react';

import MobileMenuControl from 'components/mobile-menu-control'

import { shallow } from 'enzyme';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);

describe('Testing <MobileMenuControl /> component', () => {
  const props = {
    color: 'white',
    displayMobileMenu: false,
    toggleMobileMenu: jest.fn()
  };
  const store = mockStore(props);
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MobileMenuControl store={store} />);
  });
  it('renders MobileMenuControl', () => {
    wrapper = wrapper.dive();
    expect(wrapper.exists()).toEqual(true);
  });
  it('renders MobileMenuControl with dark color', () => {
    const props = {
      color: 'dark',
      displayMobileMenu: true,
      toggleMobileMenu: jest.fn()
    };
    const store = mockStore(props);
    wrapper = shallow(<MobileMenuControl store={store} />).dive();
    expect(wrapper.exists()).toEqual(true);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
