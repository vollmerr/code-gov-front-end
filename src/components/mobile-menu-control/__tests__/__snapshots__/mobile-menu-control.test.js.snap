// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <MobileMenuControl /> component should render correctly 1`] = `
<MobileMenuControl
  color="white"
  displayMobileMenu={false}
  hideMobileMenu={[Function]}
  showMobileMenu={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  toggleMobileMenu={[Function]}
/>
`;
